routers/web/repo/view.go

```diff
@@ -232,6 +222,42 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		}
 	}
 
+	// Now limit the number of directory items to the page
+	page := ctx.QueryInt("page")
+	if page < 1 {
+		page = 1
+	}
+	pageSize := setting.UI.DirectoryPagingNum
+
+	if pageSize > 1 {
+		pager := context.NewPagination(len(entries), pageSize, page, 5)
+
+		ctx.Data["Page"] = pager
+
+		first := pageSize * (page - 1)
```

Comment:
why not use modules/util/PaginateSlice() ?

Reply:
I wasn't aware of it. I'll make a few minor improvements. (Repeatedly calling Slice is not a good idea.)